x/ecocredit/marketplace/msg_buy_direct_test.go

```diff
@@ -0,0 +1,101 @@
+package marketplace
+
+import (
+	"testing"
+
+	"gotest.tools/v3/assert"
+
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/regen-network/regen-ledger/types/math"
+)
+
+func TestMsgBuyDirect_ValidateBasic(t *testing.T) {
+	type fields struct {
+		Buyer              string
+		SellOrderId        uint64
+		Quantity           string
+		PricePerCredit     *sdk.Coin
+		DisableAutoRetire  bool
+		RetirementLocation string
+	}
+	validCoin := sdk.NewInt64Coin("ufoo", 31)
+	_, _, addr := testdata.KeyTestPubAddr()
+	tests := []struct {
+		name   string
+		fields fields
+		errMsg string
+	}{
+		{
+			name: "valid",
+			fields: fields{
+				Buyer:             addr.String(),
+				SellOrderId:       1,
+				Quantity:          "45.32",
+				PricePerCredit:    &validCoin,
+				DisableAutoRetire: true,
+			},
+		},
+		{
+			name: "valid retirement location",
+			fields: fields{Buyer: addr.String(), SellOrderId: 1, Quantity: "45", PricePerCredit: &validCoin,
+				DisableAutoRetire: false, RetirementLocation: "US-NY"},
```

Comment:
Might be nice to consistently give each field its own line as you have done in the first test case  to improve readability. Here and below, even if only one field, but ok to leave as is.

Reply:
expanded them 👍🏻 https://github.com/regen-network/regen-ledger/pull/967/commits/df038d4865e2beaabcb5a7ba0a310216b9e84535